### PR TITLE
JR2-1211 Convert html description to markdown on job ad creation

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/api/ConvertCreateJobAdvertisementDto.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package ch.admin.seco.jobs.services.jobadservice.application.jobadvertisement.dto.api;
+
+import ch.admin.seco.jobs.services.jobadservice.application.HtmlToMarkdownConverter;
+import ch.admin.seco.jobs.services.jobadservice.application.jobadvertisement.dto.create.CreateJobAdvertisementDto;
+
+public class ConvertCreateJobAdvertisementDto {
```
Comment (birom) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/api/ConvertCreateJobAdvertisementDto.java]: This name is misleading for me. I would expect a dto with a Dto suffix. Can not we simply implement the conversion logic in the JobAdvertisementRestController?

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/api/ConvertApiToNormalCreateJobAdvertisementDto.java [hunk 2]
```diff
@@ -9,7 +10,13 @@
 
 public class ConvertApiToNormalCreateJobAdvertisementDto {
```
Comment (birom) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/dto/api/ConvertApiToNormalCreateJobAdvertisementDto.java]: Please rename it to ApiToNormalCreateJobAdvertisementDtoConverter

File: infrastructure/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/markdown/RemarkHtmlToMarkdownConverter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    public String convert(String source) {
+        if (source == null) {
+            return null;
+        }
+
+        if (!containsHtmlMarkup(source)) {
+            return source;
+        }
+
+        return remark.convert(preserveLineBreaks(source));
```
Comment (birom) [inline infrastructure/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/markdown/RemarkHtmlToMarkdownConverter.java]: Why do we need this if it's been already decided that the string is a html? Doesn't this result unwanted linebreaks?

Comment (varusakov) [reply infrastructure/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/markdown/RemarkHtmlToMarkdownConverter.java]: In examples provided by Toni I've \n chars used for formatting. This line breaks will be lost during conversion to markdown if we don't replace them with `<br/>` tag.

Comment (birom) [reply infrastructure/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/markdown/RemarkHtmlToMarkdownConverter.java]: Ok, thanks. I also saw in the jira issue.

**Pull request merged**